internal/ethapi/api.go

```diff
@@ -1555,6 +1555,10 @@ func SubmitTransaction(ctx context.Context, b Backend, tx *types.Transaction) (c
 	if err := checkTxFee(tx.GasPrice(), tx.Gas(), b.RPCTxFeeCap()); err != nil {
 		return common.Hash{}, err
 	}
+	// Ensure only eip155 signed transactions are submitted
+	if !tx.Protected() {
+		return common.Hash{}, errors.New("Non-eip155 signed transaction submitted")
```

Comment:
Since this is the error message that users will see, I think we should  try to be more informative Also, we generally use lower=case in errors, since errrors are typically wrapped inside other errors. 
e.g: 
```suggestion
		return common.Hash{}, errors.New("only replay-protected (EIP-155) transactions allowed over RPC")
```